src/views/AddItem.jsx

```diff
@@ -20,27 +31,35 @@ export function AddItem({ listToken }) {
 			[name]: type === 'radio' ? parseInt(value) : value,
 		}));
 	}
-
 	// This async handleSubmit function sends the data to firebase and displays a user message
 	async function handleSubmit(event) {
 		event.preventDefault();
 		// check if user has entered an empty string or whitespace
 		if (formData.itemName.trim() === '') {
-			setSubmissionStatus('Please enter an item name.');
+			toast.error('An item name cannot be empty. Please enter an item name.');
+			return;
+		}
+		// prompts the user to remove any nonalpha-numeric characters before submission
```

Comment:
This is a great check, but I wonder if it's too strict?
For example, I can no longer add `M&M's` or `135mm 1.8 lens` to the list.

Possible bug: if I try submitting one of those above cases, it doesn't go through on the first attempt, but if I submit again, it gets added

Reply:
https://github.com/the-collab-lab/tcl-63-smart-shopping-list/assets/77761206/428b98f6-18d7-4420-be5b-744d88e323cf

Reply:
We are working on that bug, Aslan came across it as well. As far as I can tell, it is because of the toast library. Once we remove it, no problem. Our code needs a few tweaks and then it works with toast as well. But most likely we will have to remove that restriction about special chars because of the examples you mentioned.

Reply:
In the end we decided to remove the special character validation as @mentalcaries suggested because we don't want to put too much restriction.
In terms of the bug, @AslanTaheri found that it's the global search `g` that causes the regex.test function to be flipping between true and false, if we remove the g, or put the regex.test function outside of the handleSubmit, the bug is resolved.

Reply:
@shangguanwang Nice job, I'm glad it all worked out seamlessly!